File: api/baeuja/src/main/java/xyz/baeuja/api/auth/dto/SignUpRequest.java
package xyz.baeuja.api.auth.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import xyz.baeuja.api.user.domain.LoginType;
import xyz.baeuja.api.user.domain.User;

@Getter
public class SignUpRequest {

    private String email;

    private String nickname;

    @NotBlank(message = "language is required")
    private String language;

    @NotBlank(message = "timezone is required")
    private String timezone;

    @NotNull(message = "loginType is required")
    private LoginType loginType;

    public User toEntity() {
        return new User(nickname, language, timezone, loginType);
    }

    public User toEntity(String nickname) {
        return new User(nickname, language, timezone, loginType);
    }
}

File: android/baeuja/app/src/main/java/com/eello/baeuja/data/auth/dto/request/SignUpRequest.kt
package com.eello.baeuja.data.auth.dto.request

import com.eello.baeuja.domain.user.model.SignInType
import java.time.ZoneId
import java.util.Locale

data class SignUpRequest(
    val email: String? = null,
    val nickname: String? = null,
    val language: String = Locale.getDefault().language,
    val timezone: String = ZoneId.systemDefault().id,
    val loginType: SignInType
)